phoenix-core/src/main/java/org/apache/phoenix/coprocessor/CDCGlobalIndexRegionScanner.java

```diff
@@ -223,4 +172,120 @@ protected boolean getNextCoveredIndexRow(List<Cell> result) throws IOException {
+            boolean isIndexCellDeleteRow, Long indexCellTs, Cell firstCell) {
+        Map<String, Object> rowValueMap = new HashMap<>();
+
+        Map<String, Object> preImage = new HashMap<>();
+        if (this.cdcChangeScopeSet.size() == 0 ||
+                (this.cdcChangeScopeSet.contains(PTable.CDCChangeScope.PRE))) {
+            for (Map.Entry<ImmutableBytesPtr, Cell> preImageObjCell : preImageObj.entrySet()) {
+                if (dataColQualNameMap.get(preImageObjCell.getKey()) != null) {
+                    preImage.put(dataColQualNameMap.get(preImageObjCell.getKey()),
+                            dataColQualTypeMap.get(preImageObjCell.getKey()).toObject(
+                                    preImageObjCell.getValue().getValueArray()));
+                }
+            }
+            rowValueMap.put(PRE_IMAGE, preImage);
+        }
+
+        Map<String, Object> changeImage = new HashMap<>();
+        if (this.cdcChangeScopeSet.size() == 0 ||
+                (this.cdcChangeScopeSet.contains(PTable.CDCChangeScope.CHANGE))) {
+            for (Map.Entry<ImmutableBytesPtr, Cell> changeImageObjCell : changeImageObj.entrySet()) {
+                if (dataColQualNameMap.get(changeImageObjCell.getKey()) != null) {
+                    changeImage.put(dataColQualNameMap.get(changeImageObjCell.getKey()),
+                            dataColQualTypeMap.get(changeImageObjCell.getKey()).toObject(
+                                    changeImageObjCell.getValue().getValueArray()));
+                }
+            }
+            rowValueMap.put(CHANGE_IMAGE, changeImage);
+        }
+
+        Map<String, Object> postImage = new HashMap<>();
+        if (this.cdcChangeScopeSet.size() == 0 ||
+                (this.cdcChangeScopeSet.contains(PTable.CDCChangeScope.POST))){
+            if (isIndexCellDeleteRow == false) {
+                for (Map.Entry<ImmutableBytesPtr, Cell> changeImageObjCell : changeImageObj.entrySet()) {
+                    if(dataColQualNameMap.get(changeImageObjCell.getKey()) != null) {
+                        postImage.put(dataColQualNameMap.get(changeImageObjCell.getKey()),
+                                dataColQualTypeMap.get(changeImageObjCell.getKey()).toObject(
+                                        changeImageObjCell.getValue().getValueArray()));
+                    }
+                }
+                for (Map.Entry<ImmutableBytesPtr, Cell> preImageObjCell : preImageObj.entrySet()) {
+                    if(dataColQualNameMap.get(preImageObjCell.getKey()) != null
+                            && postImage.get(dataColQualNameMap.get(preImageObjCell.getKey())) == null) {
+                        postImage.put(dataColQualNameMap.get(preImageObjCell.getKey()),
+                                dataColQualTypeMap.get(preImageObjCell.getKey()).toObject(
+                                        preImageObjCell.getValue().getValueArray()));
+                    }
+                }
```

Comment:
If you flip these two, you can simply clone the `preImage` map and overwrite it with the `changeMap` which will be both simpler and faster.